core/src/components/toast/toast.tsx

```diff
@@ -175,17 +224,65 @@ export class Toast implements ComponentInterface, OverlayInterface {
     };
   }
 
+  renderButtons(buttons: ToastButton[], slot: 'start' | 'end') {
+    if (buttons.length === 0) {
+      return;
+    }
+
+    const buttonGroupsClasses = {
+      'toast-button-group': true,
+      [`toast-button-group-${slot}`]: true
+    };
+    return (
+      <div class={buttonGroupsClasses}>
+        {buttons.map(b =>
+          <button type="button" class={buttonClass(b)} tabIndex={0} onClick={() => this.buttonClick(b)}>
+            <div class="toast-button-inner">
+              {b.icon &&
+                <ion-icon
+                  icon={b.icon}
+                  lazy={false}
+                  class="toast-icon"
+                />}
```

Comment:
What's the reason for using `icon` istead of `name`, and `lazy=false` here?

I think it would also make sense to add

```
slot={!b.text ? 'icon-only' : undefined}
```